src/Integration/Binding/AdditionalFileConflictChecker.cs

```diff
@@ -0,0 +1,97 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using EnvDTE;
+
+namespace SonarLint.VisualStudio.Integration.Binding
+{
+    internal class AdditionalFileConflictChecker : IAdditionalFileConflictChecker
+    {
+        private readonly IFileSystem fileSystem;
+
+        public AdditionalFileConflictChecker()
+            : this(new FileSystem())
+        {
+        }
+
+        internal AdditionalFileConflictChecker(IFileSystem fileSystem)
+        {
+            this.fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+        }
+
+        public bool HasConflictingAdditionalFile(Project project, string additionalFileName, out string conflictingAdditionalFilePath)
+        {
+            return ExistsUnderRootFolder(project, additionalFileName, out conflictingAdditionalFilePath) ||
+                   ExistsInProject(project, additionalFileName, out conflictingAdditionalFilePath);
+        }
+
+        private bool ExistsUnderRootFolder(Project project, string additionalFileName, out string conflictingAdditionalFilePath)
+        {
+            var projectRootDirectory = Path.GetDirectoryName(project.FullName);
+            conflictingAdditionalFilePath = Path.Combine(projectRootDirectory, additionalFileName);
+
+            // For old-style MSBuild projects, the file can exist on disk but not referenced in the project, so we check using the file system
+            return fileSystem.File.Exists(conflictingAdditionalFilePath);
```

Comment:
When manually testing this case, I observed the following:
1) SonarLint.xml exists in folder but is not referenced -> detected by this method and binding fails.
2) delete the file from the disc via Windows Explorer and try again. File is not detected and binding continues as expected. However, the commit then fails with the error that a file with the same name exists.

However, it works if I close and re-open the solution after deleting the file.
It looks like the DTE hasn't picked up the fact that the file no longer exists.

It appears to be only this case that requires the solution to be reloaded.

Reply:
Hmm, this might actually be a problem, since we ask the user to delete the file and retry.
So it's very likely that this scenario will be encountered.
Perhaps I should change the message, to "Please delete the file and re-open the solution".

Reply:
Ok. Ideally we wouldn't ask the user to re-open the solution if it isn't required but that would require more work, and most users won't have an existing SonarLint.xml so it won't matter.